jdatetime/__init__.py

```diff
@@ -332,6 +332,21 @@ def fromtimestamp(timestamp):
         d = py_datetime.date.fromtimestamp(timestamp)
         (y, m, d) = GregorianToJalali(d.year, d.month, d.day).getJalaliList()
         return date(y, m, d)
+    
+    @staticmethod
+    def fromisoformat(date_string:str):
+        """
+        Convert an ISO 8601 formatted string to a date
+        :param date_string: ISO 8601 formatted string
+        """
```

Comment:
I think you can add a check here to raise an error if the `date_string` is not `str`:

```python
        if not isinstance(date_string, str):
            raise TypeError("fromisoformat: argument must be str")
```